hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -258,7 +258,7 @@ def accept(self):
         accept.__name__ = fn.__name__
         return property(accept)
 
-    def __init__(self, engine, initial, predicate):
+    def __init__(self, engine, initial, predicate, allow_transition):
```

Comment:
Suggestion: Assert that one of `predicate` or `allow_transition` be non-`None`, since if both are absent then the caller probably made a mistake.

(And it's easy enough to remove the assertion if someone genuinely wants to do that.)